napari/layers/utils/interactivity_utils.py

```diff
@@ -88,3 +93,47 @@ def drag_data_to_projected_distance(
 
     # Project the drag vector onto the specified vector(s), return the distance
     return np.einsum('j, ij -> i', drag_vector_canvas, vector).squeeze()
+
+
+def orient_plane_normal_around_cursor(layer: Image, plane_normal: tuple):
+    """Orient a rendering plane by rotating it around the cursor.
+
+    If the cursor ray does not intersect the plane, the position will remain
+    unchanged.
+
+    Parameters
+    ----------
+    layer: Image
+        The layer on which the rendering plane is to be rotated
+    plane_normal: 3-tuple
+        The target plane normal in scene coordinates.
+    """
+    # avoid circular imports
+    import napari
+
+    from ..image._image_constants import VolumeDepiction
+
+    viewer = napari.current_viewer()
+
+    # early exit
+    if viewer.dims.ndisplay != 3 or layer.depiction != VolumeDepiction.PLANE:
+        return
+
+    # find cursor-plane intersection in data coordinates
+    cursor_position = layer._world_to_displayed_data(
+        position=viewer.cursor.position, dims_displayed=layer._dims_displayed
+    )
+    view_direction = layer._world_to_data_ray(viewer.camera.view_direction)
+    intersection = layer.plane.intersect_with_line(
+        line_position=cursor_position, line_direction=view_direction
+    )
+
+    # check if intersection is within data extents for displayed dimensions
+    bounding_box = layer.extent.data[:, layer._dims_displayed]
+
+    # update plane position
+    if point_in_bounding_box(intersection, bounding_box):
+        layer.plane.position = intersection
+
+    # update plane normal
+    layer.plane.normal = layer._world_to_data_ray(plane_normal)
```

Comment:
is it possible to update the position and normal in the same call? It seems like this way there is going to be two refreshes performed.

Reply:
I had a look around the vispy image layer and I'm not actually sure where the refresh call happens

The event fires and is connected to this callback which updates the node directly
https://github.com/napari/napari/blob/a699309ffa44ac02166058706a0db4bba900ee82/napari/_vispy/layers/image.py#L198-L202

If the refresh is called once all callbacks are complete then I think it's okay, otherwise we could try something here but I'm not immediately sure how to change things given that the callbacks those events are connected to don't call refresh directly